lib/fizzy/cxx20/numeric.hpp

```diff
@@ -0,0 +1,37 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2019-2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#if __cplusplus > 201703L
+
+#include <bit>
+
+namespace fizzy
+{
+template <class T>
+constexpr int popcount(T x) noexcept
+{
+    return std::popcount.as<T>(x);
+}
+}  // namespace fizzy
+
+#else
+
+#include <cstdint>
+
+namespace fizzy
+{
+constexpr int popcount(uint32_t x) noexcept
+{
+    return __builtin_popcount(x);
```

Comment:
```suggestion
    static_assert(sizeof(x) == sizeof(unsigned int));
    return __builtin_popcount(x);
```

Reply:
There is the specific type `uint32_t x`. Why the assert?

Reply:
To make sure `uint32_t` is `unsigned int` size. The `__buintin_popcount()` takes `unsigned int`.